src/cmd-prune-containers

```diff
@@ -0,0 +1,168 @@
+    if m is None:
+        raise ValueError(f"Incorrect versioning for FCOS build {version}")
+    try:
+        timestamp = datetime.datetime.strptime(m.group(2), '%Y%m%d')
+    except ValueError:
+        raise Exception(f"FCOS build {version} has incorrect date format. It should be in (%Y%m%d)")
+    return (timestamp, int(m.group(3)))
+
+
+def get_update_graph(stream):
+
+    url = f"https://raw.githubusercontent.com/coreos/fedora-coreos-streams/main/updates/{stream}.json"
+    r = requests.get(url)
+    if r.status_code != 200:
+        raise Exception(f"Could not download update graph for {stream}. HTTP {r.status_code}")
+    return r.json()
+
+
+class BarrierRelease(Exception):
+    pass
+
+
+def main():
+
+    args = parse_args()
+
+    # Load the policy file
+    with open(args.policy, "r") as f:
+        policy = yaml.safe_load(f)
+    if args.stream not in policy:
+        raise Exception(f"Stream {args.stream} is not defined in policy file.")
+    if 'containers' not in policy[args.stream]:
+        raise Exception(f"No containers section for {args.stream} stream in policy.")
+    policy = policy[args.stream]["containers"]
+
+    print(f"Pulling tags from {args.repository_url}")
+    # This is a JSON object:
+    # {"Repository": "quay.io/jbtrystramtestimages/fcos",
+    #    "Tags": [
+    #        "40.20"40.20240301.1.0",.....]}
+    tags_cmd = subprocess.run(["skopeo", "list-tags",
+                              f"docker://{args.repository_url}"],
+                              capture_output=True)
+    tags_data = tags_cmd.stdout
+    if tags_cmd.returncode != 0:
+        raise Exception(f"Error retrieving tags list from repo: {tags_cmd.stdout}")
+
+    tags_json = json.loads(tags_data)
+    tags = tags_json['Tags']
+    # Compute the date before we should prune images
+    # today - prune-policy
+    today = datetime.datetime.now()
+    date_limit = today - relativedelta(days=convert_to_days(policy))
+    if args.v:
+        print(f"This will delete any images older than {date_limit} from the stream {args.stream}")
+
+    stream_id = STREAMS[args.stream]
+    # Get the update graph for stable streams
+    if stream_id in [1, 2, 3]:
+        update_graph = get_update_graph(args.stream)['releases']
+
+    for tag in tags:
+
+        # Process the build id and stream number
+        # TODO reuse this from  https://github.com/coreos/coreos-assembler/pull/3798/files#r1673481990
+        try:
+            (build_date, tag_stream) = parse_fcos_version(tag)
+        # ignore the named moving tags ("stable", "next" etc..)
+        except ValueError:
+            print(f"Ignoring unexpected tag: {tag}")
+            continue
+        if stream_id != tag_stream:
+            if args.v:
+                print(f"Skipping tag {tag} not in {args.stream} stream")
+            continue
+
+        # Make sure this is not a barrier release (for stable streams)
+        if stream_id in [1, 2, 3]:
```

Comment:
Hmm, that's a duplicate check. Instead of this, I think it'd be cleaner to just always initialize `update_graph` to an empty list, and then only on prod streams, we populate it. Then we can get rid of this check here since the for-loop below will automatically no-op.